### ISSUE TITLE
Ideally test that are stopped by a time-out would record a stack trace of where they were.

### ISSUE THREAD

**Issue by pcanal** [root-project/root#15144]
### Explain what you would like to see improved and how.

It would be very helpful for debugging purposes if tests that are stopped by a timeout actually print to the log file the stack trace for all threads at the time the timeout is reached. (It may or may not be 'possible' to add this).

### ROOT version

master	

### Installation method

from source

### Operating system

any

**Comment by pcanal** [comment]
Relevant discussion at https://gitlab.kitware.com/cmake/cmake/-/issues/17288

**Comment by dpiparo** [comment]
I agree this would be a nice improvement, for example to detect network glitches, deadlocks and other kind of behaviours. I wonder if this is an issue for ROOT itself, given that we cannot really have control over cmake?

**Comment by pcanal** [comment]
Given the discussion on the above mentioned link, I think we might have to implement our own solution (eg. for example have the test run a side thread that wakes up right before the timeout and print the stack track (important details in that implementation: how do we start this thread? how do we tell it the requested timeout). [I suppose it could also be a side process that send a segfault signal to the test process]

**Comment by pcanal** [comment]
The link has the solution/work around doing:
```
CTest starts the test processes via timeout -sSIGSEGV 55s {TEST_COMMAND})
```
which may also actually work.
